src/page-modules/contact/means-of-transport/means-of-transport-form-machine.ts

```diff
@@ -0,0 +1,269 @@
+import { TransportModeType } from '@atb-as/config-specs';
+import { assign, fromPromise, setup } from 'xstate';
+import { Line } from '../server/journey-planner/validators';
+import { machineEvents, Area } from '../machineEvents';
+import { InputErrorMessages, commonFieldValidator } from '../validation';
+import { convertFilesToBase64 } from '../utils';
+
+type APIParams = {
+  formType:
+    | 'driver'
+    | 'transportation'
+    | 'delay'
+    | 'stop'
+    | 'serviceOffering'
+    | 'injury'
+    | undefined;
+  area: Area | undefined;
+  transportMode: TransportModeType | undefined;
+  line: Line | undefined;
+  fromStop: Line['quays'][0] | undefined;
+  toStop: Line['quays'][0] | undefined;
+  date: string;
+  plannedDepartureTime: string;
+  feedback: string;
+  attachments?: File[];
+  firstName: string;
+  lastName: string;
+  email: string;
+};
+
+export type ContextProps = {
+  wantsToBeContacted: boolean;
+  errorMessages: InputErrorMessages;
+} & APIParams;
+
+export const meansOfTransportFormMachine = setup({
+  types: {
+    context: {} as ContextProps,
+    events: machineEvents,
+  },
+  guards: {
+    validateInputs: ({ context }) => commonFieldValidator(context),
+  },
+  actions: {
+    setFormType: assign({
```

Comment:
See other comment about using `SET_FIELD` for the form selector, but then this would not be necessary.